src/ftest.jl

```diff
@@ -229,3 +229,204 @@ function show(io::IO, ftr::FTestResult{N}) where N
+        for i = 2:size(obj.mf.f.rhs.terms[i].contrasts.matrix, 1) # correct for zero-intercept model
+            mx[i-1, :] .= mxc[i, :] - mxc[1, :]
+        end
+    else
+        mx = zeros(length(inds), p) # unknown correctness for zero-intercept model
+        for j = 1:length(inds)
+            mx[j, inds[j]] = 1
+        end
+    end
+    mx
+end
+
+tname(t::AbstractTerm) = "$(t.sym)"
+tname(t::InteractionTerm) = join(tname.(t.terms), " & ")
+tname(t::InterceptTerm) = "(Intercept)"
+
+
+
+"""
+    typeiii(obj)
+
+Calculate F-statistics for Tests of Between-Subjects Effects. 
+Sum of squares and MS not calculated.
+
+"""
+function typeiii(obj)
+    V           = vcov(obj) 
+    replace!(V, NaN => 0) # Some values can be NaN - replace it to zero
+    B           = coef(obj)   
+    c           = length(obj.mf.f.rhs.terms)
+    d           = Vector{Int}(undef, 0)
+    fac         = Vector{String}(undef, c)
+    F           = Vector{Float64}(undef,c)
+    df          = Vector{Float64}(undef, c)
+    pval        = Vector{Float64}(undef, c)
+    for i = 1:c
+        # Make L matrix
+        L       = lcontrast(obj, i)
+        #=
+        if typeof(obj.mf.f.rhs.terms[i]) <: InterceptTerm{true}
+            # I think L matrix for Intercept should represent general mean
+            # For this - L element corresponding to factor should be devided on total number of levels (not coefs)
+            # But because I can't get efficient number of levels for InteractionTerm
+            # I cant't calc it correctly
+        end
+        =#
+        if typeof(obj.mf.f.rhs.terms[i]) <: InterceptTerm{false} # If zero intercept (drop)
+            push!(d, i)
+            fac[i] = ""
+            continue
+        else
+            fac[i] = tname(obj.mf.f.rhs.terms[i])
+        end
+        # For case when cofs is zero (or NaN) we reduce rank of L-matrix
+        for c = 1:length(B)
+            if isnan(B[c]) || iszero(B[c])
+                L[:, c] .= 0
+            end
+        end
+        RL = rank(L) # Rank of L matrix
+        # F-statistics computed:
+        # F[i]    = (L'*B' * pinv(L * V * L') * L * B) / rank(L)
+        # As V is symmetric we can calc only upper triangle
+        # θ = L * V * L'
+        θ = zeros(size(L, 1), size(L, 1))
+        mulαβαtinc!(θ, L, V)
+        LB = L * B
+        # Then F can be computed:
+        # F[i]    = (LB' * pinv(Symmetric(θ)) * LB)/rank(L)
+        # I think this is more efficient:
+        F[i]    = mulαtβα(LB, pinv(Symmetric(θ))) / RL
+        df[i]   = RL
+        if iszero(df[i])
+            pval[i] = NaN
+        else
+            pval[i] = ccdf(FDist(df[i], dof_residual(obj)), F[i])
+        end
+    end
+    if length(d) > 0
+        deleteat!(fac, d)
+        deleteat!(F, d)
+        deleteat!(df, d)
+        deleteat!(pval, d)
+    end
+    GroupEffectsTable(fac, df, F, pval)
+
+end
+
+using PrettyTables
+# use PrettyTables because it is fastest way to make table
```

Comment:
I would actually use `CoefTable` for this rather than defining a new `GroupEffectsTable` type. You'd then just have `typeiii` return
```julia
CoefTable([df, F, pval], ["DF", "F", "Pr(>F)"], fac, 3, 2)
```
Also, it seems like the degrees of freedom column should be tuples with the numerator and denominator degrees of freedom for the F statistics rather than only reporting the numerator.

Reply:
done